Using central package management causes build breaking NU1009 with MAUI projects

### Description

Building a `MAUI` project using central package management does not seem possible because of the `NU1009` error. It reads as a warning, but is in fact a build breaking error.

I attached a minimal reproduction containing a `MAUI` project and a library that this project depends on, which itself requires the `ILogger` abstraction from `Microsoft.Extensions.Logging.Abstractions`,. If you build it as is, the build fails with:
```
error NU1009: The packages Microsoft.Extensions.Logging.Abstractions;Microsoft.Extensions.Logging.Abstractions;Microsoft.Extensions.Logging.Abstractions;Microsoft.Extensions.Logging.Abstractions 
 are implicitly referenced. You do not typically need to reference them from your project or in your central package versions management file. For more information, see https://aka.ms/sdkimplicitrefs
```

If you fix this error by removing the reference to `Microsoft.Extensions.Logging.Abstractions` (what the error suggests you do), the build fails with a collection of errors, due to the fact that the `ILogger` interface is cannot be found anymore in the library which the MAUI project depends on.

I also tried using https://learn.microsoft.com/en-gb/dotnet/core/project-sdk/msbuild-props#disableimplicitframeworkreferences, but it seems it is ignored altogether and has no effect.

Moreover, it doesn't seem to happen with other project types such as web API or console app.

### Steps to Reproduce

1. Clone the attached reproduction project repository
2. Run `dotnet build` at the root of the cloned repo

### Link to public reproduction project repository

https://github.com/hypdeb/repros

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Windows

### Affected platform versions

This is a build failure

### Did you find any workaround?

_No response_

### Relevant log output

_No response_

> @Cheesebaron see this workaround from @jeffkl : [#12953 (comment)](https://github.com/dotnet/maui/issues/12953#issuecomment-1409212256)

Doesn't seem to work for me.

Did this in my `Directory.Packages.Props`:

```patch
diff --git forkSrcPrefix/Directory.Packages.props forkDstPrefix/Directory.Packages.props
index c919d35848db73c4edc8d9f1a75f821f48cfd57d..4f045d5e300d9460f8cfdc112277fe1e0210d847 100644
--- forkSrcPrefix/Directory.Packages.props
+++ forkDstPrefix/Directory.Packages.props
@@ -23,7 +23,7 @@
     <PackageVersion Include="Microsoft.Extensions.Logging.Abstractions" Version="7.0.0" />
     <PackageVersion Include="Microsoft.Extensions.Logging.Console" Version="7.0.0" />
     <PackageVersion Include="Microsoft.Extensions.Options" Version="7.0.1" />
-    <PackageVersion Include="Microsoft.Maui.Graphics" Version="6.0.501" />
+    <PackageVersion Include="Microsoft.Maui.Graphics" Version="6.0.501" Condition="'$(UseMaui)' != 'true'" />
     <PackageVersion Include="Microsoft.NET.Test.Sdk" Version="17.6.0" />
     <PackageVersion Include="Microsoft.SourceLink.AzureRepos.Git" Version="1.1.1" />
     <PackageVersion Include="Microsoft.TestPlatform.ObjectModel" Version="17.0.0" />
@@ -68,10 +68,10 @@
     <PackageVersion Include="xunit" Version="2.4.2" />
     <PackageVersion Include="XunitXml.TestLogger" Version="3.0.78" />
     <PackageVersion Include="ImageCaching.Nuke" Version="3.0.0" />
-    <PackageVersion Include="Xamarin.Android.Glide" Version="4.13.2.2" />
+    <PackageVersion Include="Xamarin.Android.Glide" Version="4.13.2.2" Condition="'$(UseMaui)' != 'true'" />
     <PackageVersion Include="Xamarin.AndroidX.ConstraintLayout" Version="2.1.4.2" />
-    <PackageVersion Include="Xamarin.Google.Android.Material" Version="1.7.0.1" />
+    <PackageVersion Include="Xamarin.Google.Android.Material" Version="1.7.0.1" Condition="'$(UseMaui)' != 'true'" />
     <PackageVersion Include="TrackMan.Design.VirtualGolf" Version="0.1.10" />
     <PackageVersion Include="UXCam" Version="3.2.1" />
   </ItemGroup>
-</Project>
\ No newline at end of file
+</Project>
```

This still produces the errors:

```
/Users/tomaszcielecki/git/AppCore/TrackMan.AppCore/TrackMan.AppCore.csproj : error NU1009: The packages Microsoft.Maui.Graphics;Microsoft.Maui.Graphics;Microsoft.Maui.Graphics;Xamarin.Android.Glide are implicitly referenced. You do not typically need to reference them from your project or in your central package versions management file. For more information, see https://aka.ms/sdkimplicitrefs [/Users/tomaszcielecki/git/AppCore/TrackMan.AppCore.sln]
/Users/tomaszcielecki/git/AppCore/Playground/Playground.Droid/Playground.Droid.csproj : error NU1009: The packages Microsoft.Maui.Graphics;Xamarin.Android.Glide;Xamarin.Google.Android.Material are implicitly referenced. You do not typically need to reference them from your project or in your central package versions management file. For more information, see https://aka.ms/sdkimplicitrefs [/Users/tomaszcielecki/git/AppCore/TrackMan.AppCore.sln]
/Users/tomaszcielecki/git/AppCore/TrackMan.AppCore.TeamSports/TrackMan.AppCore.TeamSports.csproj : error NU1009: The packages Microsoft.Maui.Graphics;Microsoft.Maui.Graphics are implicitly referenced. You do not typically need to reference them from your project or in your central package versions management file. For more information, see https://aka.ms/sdkimplicitrefs [/Users/tomaszcielecki/git/AppCore/TrackMan.AppCore.sln]
```

--- a/src/Workload/Microsoft.Maui.Sdk/Sdk/BundledVersions.in.targets
+++ b/src/Workload/Microsoft.Maui.Sdk/Sdk/BundledVersions.in.targets
@@ -12,6 +12,12 @@
       however Visual Studio currently depends on this assembly being loaded for Live Visual Tree.
     -->
     <UseMauiCompat Condition=" '$(UseMauiCompat)' == '' and '$(UseMaui)' == 'true' ">true</UseMauiCompat>
+
+    <!--
+      If opted into NuGet Central Package Management, $(DisableMauiImplicitPackageReferences) defaults to true
+      https://learn.microsoft.com/nuget/consume-packages/Central-Package-Management
+    -->
+    <DisableMauiImplicitPackageReferences Condition=" '$(ManagePackageVersionsCentrally)' == 'true' and '$(DisableMauiImplicitPackageReferences)' == '' ">true</DisableMauiImplicitPackageReferences>
   </PropertyGroup>
 
   <!--
@@ -20,16 +26,16 @@
     packages. Then the packages that were specified by the user are removed from the list. Finally,
     the remaining implicit packages are added into the project.
 
-    This logic only installs packages if and only if $(EnableMauiImplicitPackageReferences) is not
+    This logic only installs packages if and only if $(DisableMauiImplicitPackageReferences) is not
     true. This property allows the user to totally override all the default nugets installed. This
     should not be needed as the $(MauiVersion) should be sufficient. However, there may be a case
     to test various CI builds of .NET MAUI.
 
     One thing to note: the old and unsupported $(UseMauiNuGets) property is the same as the new
-    $(EnableMauiImplicitPackageReferences) property and is just kept for compatibility. It should
+    $(DisableMauiImplicitPackageReferences) property and is just kept for compatibility. It should
     not be used any any project, but there are some existing projects that have used them.
   -->
-  <ItemGroup Condition=" '$(EnableMauiImplicitPackageReferences)' != 'false' and '$(UseMauiNuGets)' != 'true' ">
+  <ItemGroup Condition=" '$(DisableMauiImplicitPackageReferences)' != 'true' and '$(UseMauiNuGets)' != 'true' ">
     <_MauiImplicitPackageReference Include="Microsoft.Maui.Resizetizer" Version="$(MauiVersion)" PrivateAssets="all" Condition=" '$(UseMauiAssets)' == 'true' " />
     <_MauiImplicitPackageReference Include="Microsoft.Maui.Essentials" Version="$(MauiVersion)" Condition=" '$(UseMauiEssentials)' == 'true' ">
       <PrivateAssets Condition=" '$(OutputType)' == 'Library' and '$(AndroidApplication)' != 'true' ">all</PrivateAssets>
@@ -48,7 +54,7 @@
     </_MauiImplicitPackageReference>
   </ItemGroup>
   <!-- Install the nuget packages if we are allowed to -->
-  <ItemGroup Condition=" '$(EnableMauiImplicitPackageReferences)' != 'false' and '$(UseMauiNuGets)' != 'true' ">
+  <ItemGroup Condition=" '$(DisableMauiImplicitPackageReferences)' != 'true' and '$(UseMauiNuGets)' != 'true' ">
     <_MauiImplicitPackageReference Remove="@(PackageReference)" />
     <PackageReference Include="@(_MauiImplicitPackageReference)" IsMauiImplicitPackageReference="true" />
     <_MauiImplicitRequiredPackageReference Include="@(_MauiImplicitPackageReference->WithMetadataValue('IsMauiImplicitOptionalPackageReference',''))" />
@@ -59,7 +65,7 @@
     AfterTargets="CollectPackageReferences"
     Condition="
       ('$(SkipValidateMauiImplicitPackageReferences)' != 'true') and
-      ('$(EnableMauiImplicitPackageReferences)' != 'false' and '$(UseMauiNuGets)' != 'true') and
+      ('$(DisableMauiImplicitPackageReferences)' != 'true' and '$(UseMauiNuGets)' != 'true') and
       ('$(UseMaui)' == 'true' or '$(UseMauiCore)' == 'true' or '$(UseMauiEssentials)' == 'true' or '$(UseMauiAssets)' == 'true') and
       ('@(_MauiImplicitRequiredPackageReference->Count())' != '0') ">
 

--- a/src/Workload/README.md
+++ b/src/Workload/README.md
@@ -214,3 +214,57 @@ These folders are all .NET 6 specific, so they won't affect .NET 5 or
 older versions.
 
 After this you can install .NET 6 with a fresh install of your choice.
+
+## NuGet Central Package Management
+
+You can leverage [NuGet's central package management (CPM)][cpm] to manage all
+of your dependencies from a single location.
+
+To do this, you will need a `Directory.Packages.props` file with:
+
+```xml
+<Project>
+  <PropertyGroup>
+    <ManagePackageVersionsCentrally>true</ManagePackageVersionsCentrally>
+    <MauiVersion>8.0.0</MauiVersion>
+    <MicrosoftExtensionsVersion>8.0.0</MicrosoftExtensionsVersion>
+  </PropertyGroup>
+  <ItemGroup>
+    <PackageVersion Include="Microsoft.Maui.Core" Version="$(MauiVersion)" />
+    <PackageVersion Include="Microsoft.Maui.Controls" Version="$(MauiVersion)" />
+    <PackageVersion Include="Microsoft.Maui.Controls.Core" Version="$(MauiVersion)" />
+    <PackageVersion Include="Microsoft.Maui.Controls.Build.Tasks" Version="$(MauiVersion)" />
+    <PackageVersion Include="Microsoft.Maui.Controls.Xaml" Version="$(MauiVersion)" />
+    <PackageVersion Include="Microsoft.Maui.Essentials" Version="$(MauiVersion)" />
+    <PackageVersion Include="Microsoft.Maui.Resizetizer" Version="$(MauiVersion)" />
+    <PackageVersion Include="Microsoft.Extensions.Logging.Debug" Version="$(MicrosoftExtensionsVersion)" />
+  </ItemGroup>
+</Project>
+```
+
+For the correct value for `$(MauiVersion)` and `$(MicrosoftExtensionsVersion)`
+you will need to find a valid version number from one of:
+
+* NuGet, such as: https://www.nuget.org/packages/Microsoft.Maui.Sdk
+
+* GitHub releases, such as: https://github.com/dotnet/maui/releases
+
+Using properties like `$(MauiVersion)` and `$(MicrosoftExtensionsVersion)` are
+also completely optional, you can put the version numbers directly in the
+`%(PackageVersion.Version)` item metadata.
+
+Then in your .NET MAUI application's `.csproj` file:
+
+```xml
+<PackageReference Include="Microsoft.Maui.Core" />
+<PackageReference Include="Microsoft.Maui.Controls" />
+<PackageReference Include="Microsoft.Maui.Essentials" />
+<PackageReference Include="Microsoft.Maui.Resizetizer" />
+<PackageReference Include="Microsoft.Extensions.Logging.Debug" />
+```
+
+Note that `%(PackageReference.Version)` is intentionally left blank. See the
+documentation on [NuGet Central Package Management][cpm] for more information
+about this feature.
+
+[cpm]: https://learn.microsoft.com/nuget/consume-packages/Central-Package-Management

